engineering/documentation.md

```diff
@@ -0,0 +1,241 @@
+# Documentation at source{d}
+
```

Comment:
This doc is huge with a lot of useful resources.
Do you think a TOC would help the reader what can be found here?
I see the following:
- General Best Practices
  - State a clear goal
  - Define a scope
  - Don't alienate the reader
  - Take accessibility in mind
- Documentation Artifacts
  - README.md
  - CONTRIBUTING.md
  - Reference material
  - Release Notes
  - Issues, PullRequests, Improvement Proposals...
  - Guides/Tutorials
  - Code of Conduct
  - Other pieces documentation
- Resources

Reply:
Yeah, I'm happy to add the TOC once the content has been agreed on.